deps/rabbit/src/rabbit_vhost.erl

```diff
@@ -138,34 +139,55 @@ parse_tags(Val) when is_list(Val) ->
         [trim_tag(Tag) || Tag <- re:split(Val, ",", [{return, list}])]
     end.
 
--spec add(vhost:name(), rabbit_types:username()) -> rabbit_types:ok_or_error(any()).
-
+-spec add(vhost:name(), rabbit_types:username()) ->
+    rabbit_types:ok_or_error(any()).
 add(VHost, ActingUser) ->
-    case exists(VHost) of
-        true  -> ok;
-        false -> do_add(VHost, <<"">>, [], ActingUser)
-    end.
-
--spec add(vhost:name(), binary(), [atom()], rabbit_types:username()) -> rabbit_types:ok_or_error(any()).
+    add(VHost, #{}, ActingUser).
 
+-spec add(vhost:name(), binary(), [atom()], rabbit_types:username()) ->
+    rabbit_types:ok_or_error(any()).
 add(Name, Description, Tags, ActingUser) ->
+    add(Name, #{description => Description,
+                tags => Tags}, ActingUser).
+
+-spec add(vhost:name(), vhost:metadata(), rabbit_types:username()) ->
+    rabbit_types:ok_or_error(any()).
+add(Name, Metadata, ActingUser) ->
     case exists(Name) of
         true  -> ok;
-        false -> do_add(Name, Description, Tags, ActingUser)
+        false ->
+            catch(do_add(Name, Metadata, ActingUser))
     end.
 
-do_add(Name, Description, Tags, ActingUser) ->
+do_add(Name, Metadata, ActingUser) ->
+    Description = maps:get(description, Metadata, undefined),
+    Tags = maps:get(tags, Metadata, []),
+
+    %% validate default_queue_type
+    case Metadata of
+        #{default_queue_type := DQT} ->
+            try rabbit_queue_type:discover(DQT) of
+                _ ->
+                    ok
+            catch _:_ ->
+                      throw({error, invalid_queue_type})
+            end;
+        _ ->
+            ok
+    end,
```

Comment:
If a queue type is provided by a plugin, could we have a problem with definitions import?

Reply:
Definition import is not necessarily affected: I believe exported definitions mention the type explicitly

Reply:
Does it mean the default queue type is not stored in the exported definitions of vhosts?

Reply:
I do think `rabbitmqctl import_definitions` isn't picking this setting up

1. Create vhost-def.json containing `{"vhosts":[{"limits":[],"metadata":{"default_queue_type":"quorum","description":"","tags":[]},"name":"manual-dqt-test"}]}`
2. `rabbitmqctl import_definitions vhost-def.json
3. `rabbitmqctl ctl list_vhosts name,default_queue_type` shows all vhosts as default queue type undefined

The export / import behaviour here is a little tricky besides this though - imo the perfect outcome would be all queues were declared on import as though the client had re-sent the queue.declare method. I suspect this would require storing what the `x-queue-type` declaration property was alongside the default queue type though.

Reply:
Thanks @adamncasey. We will address such things in follow-up PRs.